xcdat/temporal.py

```diff
@@ -1679,16 +1641,11 @@ def _keep_weights(self, ds: xr.Dataset) -> xr.Dataset:
         xr.Dataset
             The dataset with the weights used for averaging.
         """
-        # Append "_original" to the name of the weights` time coordinates to
-        # avoid conflict with the grouped time coordinates in the Dataset (can
-        # have a different shape).
         if self._mode in ["group_average", "climatology"]:
-            self._weights = self._weights.rename({self.dim: f"{self.dim}_original"})
-            # Only keep the original time coordinates, not the ones labeled
-            # by group.
-            self._weights = self._weights.drop_vars(self._labeled_time.name)
+            weights = self._weights.assign_coords({self.dim: self._dataset[self.dim]})
+            weights = weights.rename({self.dim: f"{self.dim}_original"})
 
-        ds[self._weights.name] = self._weights
+        ds[weights.name] = weights
```

Comment:
Reassign the original, unlabeled time coordinates back to the `weights` xr.DataArray and then rename it to `"time_original"` to avoid conflicting the the labeled time coordinates (now called `"time"`).